pkg/storage/etcd3/watcher.go

```diff
@@ -158,13 +158,19 @@ func (wc *watchChan) sync() error {
 	wc.initialRev = getResp.Header.Revision
 
 	for _, kv := range getResp.Kvs {
-		prevResp, err := wc.watcher.client.Get(wc.ctx, string(kv.Key), clientv3.WithRev(kv.ModRevision-1), clientv3.WithSerializable())
-		if err != nil {
-			return err
-		}
 		var prevVal []byte
-		if len(prevResp.Kvs) > 0 {
-			prevVal = prevResp.Kvs[0].Value
+		if kv.ModRevision != kv.CreateRevision {
+			prevResp, err := wc.watcher.client.Get(wc.ctx, string(kv.Key), clientv3.WithRev(kv.ModRevision-1), clientv3.WithSerializable())
+			// If we can't get the previous version due to compaction, continue without a prevVal.
+			// This can resulting in an ADDED watch event being sent to a filtered watch (instead of a MODIFIED event).
```

Comment:
I still have no idea what does this mean?

So if compact happened, is it gonna be "ADDED" or "MODIFIED"?